hoomd/hpmc/update.py

```diff
@@ -894,9 +897,11 @@ def __init__(self,
 
         param_dict = ParameterDict(max_overlaps_per_particle=float,
                                    min_scale=float,
-                                   target_box=hoomd.Box,
+                                   target_box=hoomd.variant.box.BoxVariant,
```

Comment:
If this attribute is settable, we need to coerce a regular box into a box variant whenever the attribute is set, not just in the constructor.

Reply:
Yes, use a converter method here.

Reply:
I think I implemented the preprocessing correctly, but I'm running into the same import issues that I was in the vector variant PR. I will need help with that.